tdms/matlab/fdtdduration.m

```diff
@@ -6,34 +6,35 @@
 %G(t) = exp( -pi((t-to)/hwhm)^2 )
 function [to, hwhm] = fdtdduration(inputfile)
 
-[fid_input,message] = fopen(inputfile,'r');
+    [fid_input,message] = fopen(inputfile,'r');
 
-%check if file was opened successfully
-if fid_input== -1
-    error(sprintf('File %s could not be opened for reading',inputfile));
-end
-
-%proceed to_l read in config information
-current_line = fgets(fid_input);
+    %check if file was opened successfully
+    if fid_input== -1
+        error(sprintf('File %s could not be opened for reading',inputfile));
+    end
 
-while current_line ~= -1
-    eval(current_line);
+    %proceed to_l read in config information
     current_line = fgets(fid_input);
-end
 
-[epso muo c] = import_constants;
+    while current_line ~= -1
+        eval(current_line);
+        current_line = fgets(fid_input);
+    end
 
-if exist('wavelengthwidth') ~= 1
-    error('wavelengthwidth is not defined - cannot determine hwhm or to');
-end
+    [epso muo c] = import_constants;
 
-if exist('f_an') ~= 1
-    error('f_an is not defined - cannot determine hwhm or to');
-end
+    if exist('wavelengthwidth') ~= 1
+        error('wavelengthwidth is not defined - cannot determine hwhm or to');
+    end
+
+    if exist('f_an') ~= 1
+        error('f_an is not defined - cannot determine hwhm or to');
+    end
 
-refractive_index = sqrt(epsr(1));
-omega_an         = 2*pi*f_an;
-lambda_an        = c/(f_an*refractive_index);
+    refractive_index = sqrt(epsr(1));
+    omega_an         = 2*pi*f_an;
+    lambda_an        = c/(f_an*refractive_index);
 
-hwhm = lambda_an^2/((c/refractive_index)*wavelengthwidth)*2*sqrt(log(2)/pi);
-to   = hwhm*sqrt(log(1e8)/pi);
+    hwhm = lambda_an^2/((c/refractive_index)*wavelengthwidth)*2*sqrt(log(2)/pi);
+    to   = hwhm*sqrt(log(1e8)/pi);
```

Comment:
@prmunro would you mind clarifying what this variable is? Thanks!

Reply:
to (which I should have called t0) controls the time delay of the pulse that enters the FDTD/PSTD simulation when a source is introduced. In particular, it ensures that when the pulse starts to be introduced, the envelope of the pulse has a magnitude of 10^(-8). This ensures that there aren't any erroneous spectral components introduced as a result of having a sharp discontinuity in time.

Reply:
thanks – I've added a comment above the function. Will merge this PR once it has your approval

Reply:
will merge this now with a view to not have too many long-running PRs. can come back to this when people are back from holiday!